### DISCUSSION TITLE
Inconsistent Hash Values for Isomorphic Graphs Using weisfeiler_lehman_graph_hash() Across Different Versions

### DISCUSSION THREAD

**Discussion by yinkaaiwu** [networkx/networkx#7460] [Q&A]
#### Context:
I have encountered an issue with the `weisfeiler_lehman_graph_hash()` function where it probabilistically generates different hash values for the same isomorphic graph when using different versions of the library. Specifically, I noticed this issue while upgrading from version 2 to version 3.

#### Current Situation:
I am currently in the process of refactoring my code to be compatible with version 3.1 and above. My goal is to ensure that the code can handle data requirements from version 3.1 while also meeting some specific needs for version 3.3.

#### Question:
Could you please confirm if the `weisfeiler_lehman_graph_hash()` function is consistent and compatible across versions 3.1 and above? If there have been changes or improvements to this function in these versions, could you provide details or references to documentation that outline these changes?

#### Additional Information:
- **Operating System**: [Ubuntu 20.04]

Understanding the compatibility and consistency of this function across these versions is crucial for my work. I would greatly appreciate any guidance or insights on this matter.

Best regards,
[Yinkaai]

**Comment by dschult** [comment]
Can you tell us two versions that give different hashes? 
And ideally a small example graph, but if just creating random graphs will show the problem then maybe not so important).

**Reply by yinkaaiwu** [reply]
I don't quite remember which two versions of NetworkX it was. I discovered this issue a long time ago and didn't document it at the time. After finding that switching to the same version of NetworkX could solve the problem, I didn't look into it further.

**Comment by rossbar** [comment]
> Could you please confirm if the weisfeiler_lehman_graph_hash() function is consistent and compatible across versions 3.1 and above? If there have been changes or improvements to this function in these versions, could you provide details or references to documentation that outline these changes?

The best way to check for this is to look at the [release notes](https://networkx.org/documentation/latest/release/index.html). The only change to `weisfeiler_lehman_subgraph_hashes` that I noticed was #6601, which added a kwarg to allow the hashing to account for the initial node labels. AFAICT that feature addition should be backward compatible.

I just ran a basic test with `dodecahedral_graph` and can confirm that the hash is consistent across nx versions 2.7 (when the fn was added) through nx v3.3. I agree with @dschult that if you have a (preferrably as small as possible) example graph where you see this inconsistency across versions - please share it here!

Finally, a last thing to consider: are you using the same Python version when comparing nx 2.7 to nx 3.1? The `weisfeiler_lehman` functions rely on `blake2b` from `hashlib` in the standard library. Just another dimension to check to ensure this hasn't changed across Python versions.

**Reply by yinkaaiwu** [reply]
Your suggestion reminded me that the issue might be caused by an inconsistency in Python versions. I will try to find an example to reproduce the issue and share it here.

**Reply by rossbar** [reply]
Gentle ping @yinkaaiwu to see if you've found a reproducer and/or cause of the issue? If so please feel free to share; if not, perhaps the discussion can be "closed". It can always be re-opened if the issue is found to persist.
